File: test/concurrency-test/MultiLanguageTest.js
class Counter {
    constructor() {
        this.counter = 0;
    }
    
    increment() {
        this.counter = this.counter + 1; // Race condition
    }
}

class SharedData {
    constructor() {
        this.data = new Map();
    }
    
    addData(key, value) {
        this.data.set(key, value); // Race condition
    }
}

let globalVar = 0;

function worker() {
    globalVar = globalVar + 1; // Race condition
}

async function asyncTask() {
    await new Promise(resolve => setTimeout(resolve, 1000)); // Potential race condition
}

function createThread() {
    new Promise((resolve) => {
        globalVar = globalVar + 1; // Race condition
        resolve();
    });
}

const fs = require('fs');
function fileAccess() {
    fs.writeFileSync('test.txt', 'test'); // Race condition
}

module.exports = { Counter, SharedData };

File: test/concurrency-test/MultiLanguageTest.py
import threading
import asyncio

class Counter:
    def __init__(self):
        self.counter = 0
    
    def increment(self):
        self.counter = self.counter + 1  # Race condition

class SharedData:
    def __init__(self):
        self.data = {}
    
    def add_data(self, key, value):
        self.data[key] = value  # Race condition

global_var = 0

def worker():
    global global_var
    global_var = global_var + 1  # Race condition

def file_access():
    with open('test.txt', 'w') as f:  # Race condition
        f.write('test')

async def async_task():
    await asyncio.sleep(1)  # Potential race condition

def create_thread():
    threading.Thread(target=worker).start()  # Race condition

if __name__ == "__main__":
    create_thread()